src/App.js

```diff
@@ -29,13 +29,46 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
-
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
   // Wave 2
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
-  const checkForWinner = () => {
+
+  //step 7: catch the id sent in from the square component
+  //step 8: start using state and update the square info into a new squares object
+  //step 9: start a map to loop and update the info for the specific square that changes
+  //step 10: the squares update according to current player which is also a state variable defined up top
+  const onClickCallback = (id) => {
+    setSquares((squares)=>{
+      let newSquares = squares.map((square) => {
+        for(let info of square){
+          if(info.id === id && info.value === ''){
+            if(currentPlayer === PLAYER_1){
+              info.value = PLAYER_1;
+            }
+            else if(currentPlayer === PLAYER_2){
+              info.value = PLAYER_2;
+            }
```

Comment:
Why not just do this:

```suggestion
            info.value = currentPlayer;
```